puzzles/models.py

```diff
@@ -84,7 +84,7 @@ class Puzzle(SoftDeleteModel):
     class Meta:
         constraints = [
             models.UniqueConstraint(
-                fields=["name", "hunt"], name="unique_names_per_hunt"
+                fields=["name", "hunt", "deleted_at"], name="unique_names_per_hunt"
```

Comment:
Should we add a comment here explaining why this is included in the uniqueness constraint?

It isn't immediately obvious how `deleted_at` is related to `unique_names_per_hunt`.